crates/router/src/routes/user_role.rs

```diff
@@ -286,3 +286,70 @@ pub async fn list_users_in_lineage(state: web::Data<AppState>, req: HttpRequest)
     ))
     .await
 }
+
+pub async fn list_roles_with_info(state: web::Data<AppState>, req: HttpRequest) -> HttpResponse {
+    let flow = Flow::ListRolesV2;
+
+    Box::pin(api::server_wrap(
+        flow,
+        state.clone(),
+        &req,
+        (),
+        |state, user_from_token, _, _| role_core::list_roles_with_info(state, user_from_token),
+        &auth::DashboardNoPermissionAuth,
```

Comment:
`UserRead` Permission required here.

Reply:
Yes that would be more precise, for now let's keep this auth only.